src/nvim/normal.c

```diff
@@ -4437,11 +4437,8 @@ static void nv_ident(cmdarg_T *cap)
       // Start insert mode in terminal buffer
       restart_edit = 'i';
 
-      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel)<CR>", TERM_FOCUS, true);
-      do_cmdline_cmd("autocmd TermClose <buffer> "
-                     " if !v:event.status |"
-                     "   exec 'bdelete! ' .. expand('<abuf>') |"
-                     " endif");
+      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel)<bar>"
+              "bdelete!<CR>", TERM_FOCUS, true);
```

Comment:
`bdelete!` will already stop the job, so the `jobstop()` is redundant. Meaning, this can be simply `<Cmd>bdelete!<CR>`.

Reply:
Updated (813806733b8e31ef807ac8399098305a7ae2d254). Thanks!

```diff
diff --git a/src/nvim/normal.c b/src/nvim/normal.c
index f2a629417..2b5b47c0b 100644
--- a/src/nvim/normal.c
+++ b/src/nvim/normal.c
@@ -4437,8 +4437,7 @@ static void nv_ident(cmdarg_T *cap)
       // Start insert mode in terminal buffer
       restart_edit = 'i';
 
-      add_map((char_u *)"<buffer> <esc> <Cmd>call jobstop(&channel)<bar>"
-              "bdelete!<CR>", TERM_FOCUS, true);
+      add_map((char_u *)"<buffer> <esc> <Cmd>bdelete!<CR>", TERM_FOCUS, true);
     }
   }
```